### PR TITLE
Add Dialer and Logger hooks to be used in GAE

### PR DESCRIPTION
**Pull request by yzlai**
import (
    "appengine"
    "appengine/socket"
)

        ctx := appengine.NewContext(r)
        c := &smshinet.Client{
            Addr: "202.39.54.130:8000",
            Dialer: func(network, addr string) (net.Conn, error) {
                return socket.Dial(ctx, network, addr)
            },
            Logger: &smshinet.Logger{
                Debugf: func(format string, args ...interface{}){},
                Infof: func(format string, args ...interface{}){},
                Errorf: ctx.Errorf,
                Warningf: ctx.Warningf,
            },
        }

### CODE REVIEW

File: lib.go [hunk 1]
```diff
@@ -276,7 +276,7 @@ func (t *Client) sendTextNow(msg *SendMsg, recipient, message string,
 		}
 	}
 	msg.MsgSetLen = n
-	n, err = t.fillBytes(msg.MsgContent[:], message+"\x00")
+	n, err = t.fillBytes(msg.MsgContent[:], message)
```
Comment (johncylee) [inline lib.go]: Thanks for finding this, but `message+"\x00"` is not the same with `message`. `SendMsg` is designed to be reusable (although it's not used this way for now), so the ending `\0` is required. A probably better way to fix this is:
```
	// msg_type = 1、15：msg_content[ ]訊息內容的長度，不包含字串最後結尾的‘\0’符號
	msg.MsgContentLen = n - 1
```

File: lib.go [hunk 2]
```diff
@@ -120,29 +120,57 @@ var CommonRetCode map[byte]error = map[byte]error{
...
+	if t.Logger == nil {
+		t.Logger = &Logger{
+			Debugf: glog.V(1).Infof,
+			Infof: glog.Infof,
+			Errorf: glog.Errorf,
+			Warningf: glog.Warningf,
+		}
+	}
+	if t.Dialer != nil {
+        c, err := t.Dialer(network, addr)
```
Comment (johncylee) [inline lib.go]: go fmt ?

**Review by johncylee — CHANGES_REQUESTED** [review]

File: lib.go (hunk 1 above)
Comment (yzlai) [reply lib.go]: Actually, []byte is default filled by \x00.
Anyway, I'll update as you wish.

Comment (johncylee) [reply lib.go]: I know, but not if we want to re-use it. :)  Thank you!

**Comment by yzlai** [comment]
amend force pushed, please review, thanks!

**Review by johncylee — APPROVED** [review]

**Pull request merged**